### PR TITLE
fix: restart hooks server, ensure __wg exists

### PR DESCRIPTION
**Pull request by StarpTech**
#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: cli/commands/up.go [hunk 1]
```diff
@@ -205,39 +131,77 @@ var upCmd = &cobra.Command{
...
-				}
-			}()
 
-			hooksBundler.BundleAndWatch(ctx)
+			onAfterBuild = func() {
+				log.Debug("Configuration change detected")
+				// generate new config
+				<-configRunner.Run(ctx)
+
+				// bundle server
```
Comment (jensneuse) [inline cli/commands/up.go:147]: bundle hooks

File: cli/commands/up.go [hunk 2]
```diff
@@ -205,39 +131,77 @@ var upCmd = &cobra.Command{
...
+				hooksBundler.Bundle()
+				go func() {
+					// run or restart server
+					<-hookServerRunner.Run(ctx)
```
Comment (jensneuse) [inline cli/commands/up.go:151]: does Run kill and restart automatically?

File: cli/commands/up.go [hunk 3]
```diff
@@ -268,8 +232,18 @@ var upCmd = &cobra.Command{
 			log.Info("Received interrupt signal. Initialize WunderNode shutdown ...",
 				abstractlogger.String("signal", signal.String()),
 			)
+			configRunner.Stop()
```
Comment (jensneuse) [inline cli/commands/up.go:235]: Can you not just move the Stop logic below the select case? no need to duplicate it I think. select case will block until one of signal or ctx done returns. So this logic can go above n.Close (249)

File: packages/sdk/src/middleware/server.ts [hunk 4]
```diff
@@ -193,19 +191,19 @@ export const startServer = async (
...
 			req.ctx = {
 				log: req.log.child({ plugin: 'hooks' }),
-				user: req.body.__wg.user,
+				user: req.body.__wg?.user,
 				// clientRequest represents the original client request that was sent initially to the server.
 				clientRequest: {
 					headers: new Headers(req.body.__wg.clientRequest?.headers),
-					requestURI: req.body.__wg.clientRequest?.requestURI || '',
-					method: req.body.__wg.clientRequest?.method || 'GET',
+					requestURI: req.body.__wg?.clientRequest?.requestURI || '',
```
Comment (jensneuse) [inline packages/sdk/src/middleware/server.ts:203]: why __wg? when the definition in line 194 says that __wg will always be set? Make it optional in 194 or remove the '?'?

File: pkg/apihandler/apihandler.go [hunk 5]
```diff
@@ -2114,7 +2114,7 @@ func getOperationMetaData(r *http.Request) *OperationMetaData {
 
 func hookBaseData(r *http.Request, buf []byte, variables []byte, response []byte) []byte {
 	buf = buf[:0]
-	buf = append(buf, []byte("{}")...)
+	buf = append(buf, []byte(`{ "__wg": {} }`)...)
```
Comment (jensneuse) [inline pkg/apihandler/apihandler.go:2117]: no need for whitespace in JSON

File: pkg/bundler/bundler.go [hunk 6]
```diff
@@ -41,6 +41,7 @@ type Config struct {
 	WatchPaths            []string
 	IgnorePaths           []string
 	OutFile               string
+	OnAfter               func()
```
Comment (jensneuse) [inline pkg/bundler/bundler.go:44]: Can we be a bit more specific here? e.g. OnAfterBuild if that's the intention?

File: pkg/scriptrunner/scriptrunner.go [hunk 7]
```diff
@@ -87,6 +87,14 @@ func (b *ScriptRunner) Run(ctx context.Context) chan struct{} {
 			// exit code == -1 means the script was killed by a signal
 			// this is intentional and not an error and happens when we re-start the process after a watched file has changed
 			if status.Exit == -1 {
+				b.log.Error("Script exited",
```
Comment (jensneuse) [inline pkg/scriptrunner/scriptrunner.go:90]: Should this be logged as an error? It's intentional. Maybe Debug?

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: cli/commands/up.go (hunk 2 above)
Comment (StarpTech) [reply cli/commands/up.go:151]: Yes

File: pkg/scriptrunner/scriptrunner.go (hunk 7 above)
Comment (StarpTech) [reply pkg/scriptrunner/scriptrunner.go:90]: I wanted to handle it as debug. Good catch.

File: cli/commands/up.go (hunk 3 above)
Comment (StarpTech) [reply cli/commands/up.go:235]: I like.

**Review by jensneuse — APPROVED** [review]

**Pull request merged**